consensus/integration/network_test.go

```diff
@@ -124,7 +126,9 @@ func (n *Network) unicast(event interface{}, channel network.Channel, targetID f
 	}
 
 	sender, receiver := n.node, net.node
+	n.hub.lock.Lock()
 	block, delay := n.hub.filter(channel, event, sender, receiver)
+	n.hub.lock.Unlock()
```

Comment:
instead of putting the locking in the higher-level code, wouldn't it make sense if the implementations of `BlockOrDelayFunc` were itself concurrency safe?

Reply:
+1 I agree, and seems like most of the implementations are already concurrency safe